src/java.desktop/share/classes/java/awt/image/ComponentSampleModel.java

```diff
@@ -118,19 +118,16 @@
      * @param scanlineStride the line stride of the region of image
      *     data described
      * @param bandOffsets the offsets of all bands
-     * @throws IllegalArgumentException if {@code w} or
-     *         {@code h} is not greater than 0
-     * @throws IllegalArgumentException if {@code pixelStride}
-     *         is less than 0
-     * @throws IllegalArgumentException if {@code scanlineStride}
-     *         is less than 0
-     * @throws IllegalArgumentException if {@code numBands}
-     *         is less than 1
+     * @throws IllegalArgumentException if {@code w} and {@code h}
+     *         are not both greater than 0
      * @throws IllegalArgumentException if the product of {@code w}
-     *         and {@code h} is greater than
-     *         {@code Integer.MAX_VALUE}
+     *         and {@code h} is greater than {@code Integer.MAX_VALUE}
+     * @throws IllegalArgumentException if {@code pixelStride} is less than 0
+     * @throws IllegalArgumentException if {@code scanlineStride} is less than 0
+     * @throws NullPointerException if {@code bandOffsets} is {@code null}
```

Comment:
I like the NPE here, but probably it makes sense to use the same exception for all errors? "IllegalArgumentException "

Reply:
Meaning even though you like the NPE you would change it to IAE like the test ?

Reply:
Yes and no, I just wanted to highlight this place, since somebody may prefer IAE here.